clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleZonalServerGroupCachingAgent.java

```diff
@@ -0,0 +1,906 @@
+    serverGroup.setInstances(instances);
+    serverGroup.setNamedPorts(convertNamedPorts(manager));
+    serverGroup.setSelfLink(manager.getSelfLink());
+    serverGroup.setCurrentActions(manager.getCurrentActions());
+
+    setLaunchConfig(serverGroup, manager, instanceTemplate, credentials, providerCache);
+    setAutoscalerGroup(serverGroup, manager, instanceTemplate, objectMapper);
+    if (instanceTemplate != null) {
+      InstanceProperties properties = instanceTemplate.getProperties();
+      if (properties != null) {
+        serverGroup.setCanIpForward(properties.getCanIpForward());
+        if (properties.getServiceAccounts() != null) {
+          serverGroup.setInstanceTemplateServiceAccounts(
+              ImmutableSet.copyOf(properties.getServiceAccounts()));
+        }
+        if (properties.getTags() != null && properties.getTags().getItems() != null) {
+          serverGroup.setInstanceTemplateTags(ImmutableSet.copyOf(properties.getTags().getItems()));
+        }
+        if (properties.getLabels() != null) {
+          serverGroup.setInstanceTemplateLabels(ImmutableMap.copyOf(properties.getLabels()));
+        }
+        if (properties.getNetworkInterfaces() != null
+            && !properties.getNetworkInterfaces().isEmpty()
+            && properties.getNetworkInterfaces().get(0) != null) {
+          serverGroup.setNetworkName(
+              Utils.decorateXpnResourceIdIfNeeded(
+                  credentials.getProject(), properties.getNetworkInterfaces().get(0).getNetwork()));
+        }
+      }
+    }
+    serverGroup.setStatefulPolicy(manager.getStatefulPolicy());
+    if (manager.getAutoHealingPolicies() != null && !manager.getAutoHealingPolicies().isEmpty()) {
+      serverGroup.setAutoHealingPolicy(manager.getAutoHealingPolicies().get(0));
+    }
+    populateAutoscaler(serverGroup, autoscaler);
+    return serverGroup;
+  }
+
+  private static void setRegionConfig(
+      GoogleServerGroup serverGroup,
+      InstanceGroupManager manager,
+      GoogleNamedAccountCredentials credentials) {
+
+    serverGroup.setRegional(manager.getZone() == null);
+
+    if (serverGroup.getRegional()) {
+      serverGroup.setRegion(Utils.getLocalName(manager.getRegion()));
+      List<String> zones = getZones(manager.getDistributionPolicy());
+      serverGroup.setZones(ImmutableSet.copyOf(zones));
+      serverGroup.setDistributionPolicy(new GoogleDistributionPolicy(zones));
+    } else {
+      String zone = Utils.getLocalName(manager.getZone());
+      serverGroup.setZone(zone);
+      serverGroup.setZones(ImmutableSet.of(zone));
+      serverGroup.setRegion(credentials.regionFromZone(zone));
+    }
+  }
+
+  private static List<String> getZones(DistributionPolicy distributionPolicy) {
+    if (distributionPolicy == null || distributionPolicy.getZones() == null) {
+      return ImmutableList.of();
+    }
+    return distributionPolicy.getZones().stream()
+        .map(z -> Utils.getLocalName(z.getZone()))
```

Comment:
Maybe more stream-y as:
```
      .map(DistributionPolicyZoneConfiguration::getZone)
      .map(Utils::getLocalName)
```
🤷‍♂

Reply:
I dunno, I'm generally of the mind that merging consecutive map calls aids readability instead of hinders it? (Obviously up to some limit, where the lambda is still relatively readable.)

IOW: I don't think that's necessary.